code/datums/elements/action_state/move_to_atom.dm

```diff
@@ -6,45 +6,46 @@
 	var/list/stutter_step_prob = list() //The prob() chance of a mob going left or right when distance is maintained with the target
 
 /datum/element/action_state/move_to_atom/process()
-	for(var/mob/living/carbon/mob in distances_to_maintain)
-		if(!mob.canmove || mob.stat == DEAD)
+	for(var/mob_to_process in distances_to_maintain)
+		if(!mob_to_process.canmove || mob_to_process.stat == DEAD)
```

Comment:
This will need typecasting. Outside of the for loop if there can only be mobs in `distances_to_maintain`, inside else.